### PR TITLE
chore(doc): Update Readme.md

### PR DESCRIPTION
**Pull request by yuxizama**
- Reorganized existing content to follow the new structure
- Aligned all the header links and badges
- Added a table of contents
- Added a call-to-action for users to star the repo
- Checked all links and replaced outdated ones.
- Unified the formatting
- Updated the support banner 
- Improved the overall content quality
- Added missing content where necessary
- Added more “resources” for each product
- Ensured wording is consistent throughout

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,34 +163,63 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 To run this code, use the following command: 
 <p align="center"> <code> cargo run --release </code> </p>
 
-Note that when running code that uses `tfhe-rs`, it is highly recommended
+> ℹ️ Note that when running code that uses `tfhe-rs`, it is highly recommended
 to run in release mode with cargo's `--release` flag to have the best performances possible.
 
+*This example is explained in more detail in [this part of the documentation](https://docs.zama.ai/tfhe-rs/getting-started/quick_start)*
```
Comment (IceTDrinker) [inline README.md:170]: It's not the same example, so I would say

```
Find a more detailed example in ...
```

Or an example with more explanation in...

File: README.md [hunk 2]
```diff
@@ -120,34 +163,63 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
-Only approved contributors can send pull requests, so please make sure to get in touch before you do!
+## Resources 
 
-## Credits
+### 🟨 TFHE deep dive
+- [TFHE Deep Dive - Part I - Ciphertext types](https://www.zama.ai/post/tfhe-deep-dive-part-1)
+- [TFHE Deep Dive - Part II - Encodings and linear leveled operations](https://www.zama.ai/post/tfhe-deep-dive-part-2)
+- [TFHE Deep Dive - Part III - Key switching and leveled multiplications](https://www.zama.ai/post/tfhe-deep-dive-part-3)
+- [TFHE Deep Dive - Part IV - Programmable Bootstrapping](https://www.zama.ai/post/tfhe-deep-dive-part-4)
+<br></br>
```
Comment (IceTDrinker) [inline README.md:184]: what are the br for ?

Not a pro when it comes to html tags

File: README.md [hunk 3]
```diff
@@ -120,34 +163,63 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
-## Need support?
-<a target="_blank" href="https://community.zama.ai">
-  <img src="https://github.com/zama-ai/tfhe-rs/assets/157474013/33d856dc-f25d-454b-a010-af12bff2aa7d">
-</a>
 
+*Explore more useful resources in [TFHE-rs tutorials](https://docs.zama.ai/tfhe-rs/tutorials) and [Awesome Zama repo](https://github.com/zama-ai/awesome-zama)*
+<br></br>
+### 🟨 Documentation
 
+Full, comprehensive documentation is available here: [https://docs.zama.ai/concrete-ml](https://docs.zama.ai/tfhe-rs).
```
Comment (IceTDrinker) [inline README.md:198]: link is wrong, referencing concrete-ml

File: README.md [hunk 4]
```diff
@@ -4,40 +4,82 @@
...
+  - [What is TFHE-rs](#-what-is-tfhe-rs)
+  - [Main features](#-main-features)
+- **[Getting Started](#getting-started)**
+   - [Configuration](#-configuration)
+   - [A simple example](#-a-simple-example)
+- **[Resources](#resources)**
+   - [TFHE deep dive](#-tfhe-deep-dive)
+   - [Tutorials](#-tutorials)
+   - [Documentation](#-documentation)
+- **[Working with THFE-rs](working-with-tfhe-rs)**
```
Comment (IceTDrinker) [inline README.md:37]: typo THFE -> TFHE

File: README.md [hunk 5]
```diff
@@ -4,40 +4,82 @@
...
 implementation. The goal is to have a stable, simple, high-performance, and
 production-ready library for all the advanced features of TFHE.
+<br></br>
+
+### 🟨 Main features
+
+- **Low-level cryptographic library:** implementing Zama’s variant of TFHE, including programmable bootstrapping
+- **Implementation of the original TFHE boolean API:** used as a drop-in replacement for other TFHE libraries
+- **Short integer API:**  enabling exact, unbounded FHE integer arithmetics with up to 8 bits of message space
+- **Public key encryption:** implementation for TFHE
```
Comment (IceTDrinker) [inline README.md:68]: this looks odd, the content after the `:` is not really useful

Though we could say we have a Size efficient Public Key encryption

File: README.md [hunk 6]
```diff
@@ -4,40 +4,82 @@
...
+- **Ciphertext and server key compression** : for efficient data transfer
```
Comment (IceTDrinker) [inline README.md:69]: same here, we could probably drop the `:` everywhere

File: README.md [hunk 7]
```diff
@@ -4,40 +4,82 @@
...
+- **Full Rust API, C bindings to the Rust High-Level API, and client-side Javascript API using WASM**.
+
+*Learn more features in TFHE-rs's [documentation](https://docs.zama.ai/tfhe-rs/readme).*
```
Comment (IceTDrinker) [inline README.md:72]: this sentence looks a bit strange 🤔

File: README.md [hunk 8]
```diff
@@ -120,34 +163,63 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 To run this code, use the following command: 
 <p align="center"> <code> cargo run --release </code> </p>
 
-Note that when running code that uses `tfhe-rs`, it is highly recommended
+> ℹ️ Note that when running code that uses `tfhe-rs`, it is highly recommended
```
Comment (IceTDrinker) [inline README.md:166]: tfhe-rs -> TFHE-rs I guess to stick with the branding, seems it's an old mistake

File: README.md [hunk 9]
```diff
@@ -159,22 +231,36 @@ To cite TFHE-rs in academic papers, please use the following entry:
...
 
-When a new update is published in the Lattice Estimator, we update parameters accordingly.
 
-### Side-Channel Attacks
+## Support
 
-Mitigation for side channel attacks have not yet been implemented in TFHE-rs,
-and will be released in upcoming versions.
+<a target="_blank" href="https://community.zama.ai">
+  <img src="https://github.com/zama-ai/concrete-ml/assets/157474013/9d518f66-30da-4738-a154-a4d9fce93704">
```
Comment (IceTDrinker) [inline README.md:254]: using the concrete-ml assets ?

**Review by IceTDrinker — COMMENTED** [review]
Some things I saw when going over, thanks

File: README.md [hunk 10]
```diff
@@ -4,40 +4,82 @@
...
   </a>
 </p>
-<hr/>
 
 
-**TFHE-rs** is a pure Rust implementation of TFHE for boolean and integer
-arithmetics over encrypted data. It includes:
- - a **Rust** API
- - a **C** API
- - and a **client-side WASM** API
```
Comment (yuxizama) [inline README.md:30]: Aren't they repetitive with the last points of the main features? "Full Rust API, C bindings to the Rust High-Level API, and client-side Javascript API using WASM"

File: README.md (hunk 1 above)
Comment (yuxizama) [reply README.md]: ✅

**Review by yuxizama — COMMENTED** [review]
Thanksfor the feedback! 
I've adjusted accordingly and changed some minor details.

File: README.md (hunk 9 above)
Comment (IceTDrinker) [reply README.md:251]: no I'm fine with it as long as it does not break !

File: README.md (hunk 5 above)
Comment (IceTDrinker) [reply README.md]: nope all good

File: README.md (hunk 10 above)
Comment (IceTDrinker) [reply README.md:30]: I think it's fine ?

**Review by IceTDrinker — APPROVED** [review]
Good to merge this first set of changes, thanks !

**Comment by IceTDrinker** [comment]
> I've made some last few commits. Could you take care of the merge? @IceTDrinker Thanks!

Will do, if the content is ok I'll do the git house keeping on my end

**Pull request merged**